go-controller/pkg/ovn/libovsdbops/model.go

```diff
@@ -0,0 +1,239 @@
+package libovsdbops
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/ovn-org/libovsdb/model"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/sbdb"
+)
+
+func getUUID(model model.Model) string {
+	switch t := model.(type) {
+	case *nbdb.ACL:
+		return t.UUID
+	case *nbdb.AddressSet:
+		return t.UUID
+	case *nbdb.BFD:
+		return t.UUID
+	case *nbdb.GatewayChassis:
+		return t.UUID
+	case *nbdb.LoadBalancer:
+		return t.UUID
+	case *nbdb.LogicalRouter:
+		return t.UUID
+	case *nbdb.LogicalRouterPolicy:
+		return t.UUID
+	case *nbdb.LogicalRouterPort:
+		return t.UUID
+	case *nbdb.LogicalRouterStaticRoute:
+		return t.UUID
+	case *nbdb.LogicalSwitch:
+		return t.UUID
+	case *nbdb.LogicalSwitchPort:
+		return t.UUID
+	case *nbdb.NAT:
+		return t.UUID
+	case *nbdb.PortGroup:
+		return t.UUID
+	case *sbdb.Chassis:
+		return t.UUID
+	default:
+		panic(fmt.Sprintf("getUUID: unknown model %T", t))
+	}
+}
+
+func setUUID(model model.Model, uuid string) {
+	switch t := model.(type) {
+	case *nbdb.ACL:
+		t.UUID = uuid
+	case *nbdb.AddressSet:
+		t.UUID = uuid
+	case *nbdb.BFD:
+		t.UUID = uuid
+	case *nbdb.GatewayChassis:
+		t.UUID = uuid
+	case *nbdb.LoadBalancer:
+		t.UUID = uuid
+	case *nbdb.LogicalRouter:
+		t.UUID = uuid
+	case *nbdb.LogicalRouterPolicy:
+		t.UUID = uuid
+	case *nbdb.LogicalRouterPort:
+		t.UUID = uuid
+	case *nbdb.LogicalRouterStaticRoute:
+		t.UUID = uuid
+	case *nbdb.LogicalSwitch:
+		t.UUID = uuid
+	case *nbdb.LogicalSwitchPort:
+		t.UUID = uuid
+	case *nbdb.NAT:
+		t.UUID = uuid
+	case *nbdb.PortGroup:
+		t.UUID = uuid
+	case *sbdb.Chassis:
+		t.UUID = uuid
+	default:
+		panic(fmt.Sprintf("setUUID: unknown model %T", t))
+	}
+}
+
+func copyIndexes(model model.Model) model.Model {
```

Comment:
hm.... is there a way to get the indexes fromm the model? I am afraid we will forget to update this as the schema/model evolves.

Reply:
There's no way. I thought about keeping the deep copy that Alex had. But then again, there's no problem if we miss indexes we don't intend to use. And if we do use them, then some new functionality will never work, it's not like it will start failing from nowhere.

Reply:
ack. no problem.